### ISSUE TITLE
support jdk 1.7

### ISSUE THREAD

**Issue by jorgheymans** [benas/random-beans#49]
Hi,

Nice library ! I was hoping to use it but sadly we are still on 1.7 
Would it not be possible to keep support for 1.7 whilst adding 1.8 features ?

Thanks
Jorg

**Comment by benas** [comment]
Hello,

> Nice library !

Thank you.

> Would it not be possible to keep support for 1.7 whilst adding 1.8 features ?

Yes, the library requires java 1.8 to compile and Java 1.6 to run (only the `random-beans-java8` module requires java 8). see [here](https://github.com/benas/random-beans/issues/28#issuecomment-166434694)

Regards
Mahmoud

**Comment by heymjo** [comment]
Thanks for the reply. Could it be that the compiler level is not set correctly then ? Because i got a class number exception when trying to run it in my 1.7 project. IIRC something like below ?

```
      <plugin>
        <artifactId>maven-compiler-plugin</artifactId>
        <configuration>
          <source>1.8</source>
          <target>1.6</target>
        </configuration>
      </plugin>
```

**Comment by benas** [comment]
This is what I said I tried [here](https://github.com/benas/random-beans/issues/28#issuecomment-166434694) and in more details [here](https://github.com/benas/random-beans/issues/41#issuecomment-166571372).. But it is not working, I got a compilation error:

```
[INFO] Compiling 75 source files
javacTask: source release 1.8 requires target release 1.8
```

The same for target version 1.7.

Could you please give me a stack trace or the exact compilation error?
Thank you

**Comment by heymjo** [comment]
Using jdk1.7.0_79-64bit, my unit test:

```java
        PopulatorBuilder populatorBuilder = PopulatorBuilder.aNewPopulator();
        MyDto dto = populatorBuilder.build().populateBean(MyDto.class);
```

getting this:

```
java.lang.UnsupportedClassVersionError: io/github/benas/randombeans/PopulatorBuilder : Unsupported major.minor version 52.0

	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at eu.europa.ec.agri.cat.cepflow.internal.db.CepFlowDaoTest.testPopulator(CepFlowDaoTest.java:46)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
```

**Comment by benas** [comment]
Hi,

I've decided to postpone java 8 support to version 3 of the library. Here is the required Java version for each release:

- jPopulator 1.x (current) : Java 6
- Random Beans 2.x (next) : Java 7
- Random Beans 3.x (next) : Java 8

I've fixed this issue and deployed the 2.0.0-SNAPSHOT version to maven central.

You can give it a try with Java 7 and keep me informed in case of any issue.

Kind regards
Mahmoud
